compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5384,6 +5389,37 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
         }
     }
 
+    private void checkArrayLibSortFuncArgs(BLangInvocation iExpr) {
+        if (iExpr.argExprs.size() <= 2 && !types.isOrderedType(iExpr.expr.type)) {
+            dlog.error(iExpr.expr.pos, DiagnosticCode.INVALID_SORT_ARRAY_MEMBER_TYPE, iExpr.expr.type);
+        }
+
+        if (iExpr.argExprs.size() != 3) {
+            return;
```

Comment:
Don't we need any error message for this case and it needs to be the first condition to verify in the method.

Reply:
`iExpr.argExprs` list can have 1 to a maximum of 3  members considering the scenarios given below
arr.sort(), arr.sort(direction), arr.sort(direction, key). Therefore this can not be taken as the first condition to be verified. 
When there are more than 3 arguments it'll be handled by the following https://github.com/ballerina-platform/ballerina-lang/blob/8481a2bd849a79154b46e06d56d3557762ce64c5/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L5299